pkg/storage/wal/manager.go

```diff
@@ -185,9 +191,25 @@ func (m *Manager) Append(r AppendRequest) (*AppendResult, error) {
 	return it.r, nil
 }
 
+func (m *Manager) Close() {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	m.closed = true
+	if m.available.Len() > 0 {
```

Comment:
This is duplicate code. I will refactor it in a follow up PR. But I wanted to show how it works here.